pages/admin/request/[requestId].tsx

```diff
@@ -0,0 +1,220 @@
+  applicant: {
+    id: 1,
+    rcdUserId: 1,
+    mostRecentAPP: 12345,
+    firstName: 'Applicant',
+    lastName: 'One',
+    gender: Gender.Female,
+    dateOfBirth: 1,
+    email: 'applicantone@email.com',
+    phone: '1234567890',
+    province: Province.Bc,
+    city: 'Richmond',
+    addressLine1: '123 Richmond St.',
+    postalCode: 'X0X0X0',
+  },
+  physician: {
+    id: 1,
+    mspNumber: 123456789,
+    firstName: 'Physician',
+    lastName: 'One',
+    phone: '1234567890',
+    addressLine1: '123 Richmond St.',
+    postalCode: 'X0X0X0',
+    city: 'Richmond',
+    province: Province.Bc,
+    status: PhysicianStatus.Active,
+  },
+  replacement: {
+    // Data will be queried from replacements table using relation from Application - Replacement
+    cause: 'Stolen by goose',
+    timestamp: new Date().toDateString(),
+    locationLost: 'Waterloo, ON',
+    description: 'A goose stole it from my wallet.',
+    isUpdated: true,
+  },
+  createdAt: new Date().toDateString(),
+  expirationDate: new Date().toDateString(),
+  isRenewal: true,
+  applicationStatus: 'PENDING' as ApplicationStatus, // Will be updated in the future as we'll be able to reference ApplicationStatus enum in types.ts
+  permitFee: 5,
+  donation: 10,
+  paymentType: PaymentType.Visa,
+  applicationProcessingStepsCompleted: [] as number[], // The format of this will change. For now, 6 steps is complete.
+};
+
+// Individual request page
+export default function Request() {
+  const [application, setApplication] = useState(mockApplication);
+
+  const {
+    applicant,
+    physician,
+    replacement,
+    createdAt,
+    expirationDate,
+    isRenewal,
+    applicationStatus,
+    permitFee,
+    donation,
+    paymentType,
+    applicationProcessingStepsCompleted,
+  } = application;
+
+  // Approve modal
+  const onApprove = () => {
+    // TODO: Make mutation call to modify Application's status to ApplicationStatus.INPROGRESS
+    setApplication({ ...application, applicationStatus: 'INPROGRESS' });
+  };
+
+  // Reject modal
+  const onReject = () => {
+    // TODO: Make mutation call to modify Application's status to ApplicationStatus.REJECTED
+    setApplication({ ...application, applicationStatus: 'REJECTED' });
+  };
+
+  // Edit personal information modal
+  const onComplete = () => {
+    // TODO: Make mutation call to modify Application's status to ApplicationStatus.COMPLETED
+    setApplication({ ...application, applicationStatus: 'COMPLETED' });
+  };
+
+  // Edit doctor information/reason for replacement modal
+
+  // Edit payment information modal
+
+  // Processing tasks completion handler
+  const onTaskComplete = (taskNum: number) => {
+    // TODO: Update the appropriate table
+    setApplication({
+      ...application,
+      applicationProcessingStepsCompleted: [...applicationProcessingStepsCompleted, taskNum],
+    });
+  };
+
+  // Removes the taskNum from the list of completed tasks
+  const onTaskUndo = (taskNum: number) => {
```

Comment:
Note: Right now, this doesn't clear the assigned APP number or invoice number. When you work on the API hookup, we should keep that in mind. No action is required as of now